packages/components/src/components/button/button.tsx

```diff
@@ -87,20 +89,42 @@ export class Button {
   /** Add ARIA attributes. */
   @Prop() public aria?: SelectedAriaAttributes<ButtonAriaAttribute>;
 
+  /** The id of a form element the button should be associated with. */
+  @Prop({ reflect: true }) public form?: string;
+  // In the React wrapper, all props are synced as properties on the element ref, so reflecting "form" as an attribute ensures it is properly handled in the form submission process.
+
+  @AttachInternals() private internals: ElementInternals;
+
   private initialLoading: boolean = false;
 
   @Listen('click', { capture: true })
   public onClick(e: MouseEvent): void {
     if (isDisabledOrLoading(this.disabled, this.loading)) {
       e.stopPropagation();
+      return;
     }
+
+    if (this.form && this.internals.form) {
```

Comment:
We also want to utilise `ElementInternals` when used inside of a form without the form attribute. Currently this is achieved by the `improveButtonHandlingForCustomElement ` function